src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,468 @@
+
+                burst_map['burst_id'].append(burst.burst_id)
+                # keep the burst object so we don't have to re-parse
+                burst_map['burst'].append(burst)
+
+                left, bottom, right, top = bbox_utm
+                burst_map['x_top_left'].append(left)
+                burst_map['y_top_left'].append(top)
+                burst_map['x_bottom_right'].append(right)
+                burst_map['y_bottom_right'].append(bottom)
+
+                burst_map['epsg'].append(epsg)
+                burst_map['date'].append(burst.sensing_start.strftime("%Y%m%d"))
+                # Save the file paths for creating the runconfig
+                burst_map['orbit_path'].append(orbit_path)
+                burst_map['zip_file'].append(zip_file)
+
+    burst_map = pd.DataFrame(data=burst_map)
+    return burst_map
+
+
+def snap_geogrid(geogrid, x_spac, y_spac):
+    """Snap the bounding box of the geogrid to integer multiples of the
+    spacing in X-/Y-directions.
+
+    Parameters
+    ----------
+    geogrid : isce3.core.Geogrid
+        Geogrid object
+    x_spac : float
+        Spacing of geocoded burst along X-direction
+    y_spac : float
+        Spacing of geocoded burst along Y-direction
+
+    Returns
+    -------
+    tuple
+        Tuple containing the snapped bounding box coordinates
+        (left, bottom, right, top)
+    """
+    x_left = x_spac * np.floor(geogrid.start_x / x_spac)
+    x_right = x_spac * np.ceil((geogrid.start_x + x_spac * geogrid.width) / x_spac)
+    y_bottom = y_spac * np.floor((geogrid.start_y + y_spac * geogrid.length) / y_spac)
+    y_top = y_spac * np.ceil(geogrid.start_y / y_spac)
+    return (x_left, y_bottom, x_right, y_top)
+
+
+
+
+
+def prune_dataframe(data, id_col, id_list):
+    '''
+    Prune dataframe based on column ID and list of value
+    Parameters:
+    ----------
+    data: pandas.DataFrame
+        dataframe that needs to be pruned
+    id_col: str
+        column identification for 'data' (e.g. 'burst_id')
+    id_list: list
+        List of elements to consider when pruning.
+        If exclude_items is False (default), then all elements in `data`
+            will be kept *except for* those in `id_list`.
+        If exclude_items is True, the items in `id_list` will be removed from `data`.
+    exclude_items: bool
+        If True, the items in `id_list` will be removed from `data`.
+        If False, all elements in `data` will be kept *except for* those in `id_list`.
+    Returns:
+    -------
+    data: pandas.DataFrame
+       Pruned dataframe with rows in 'id_list'
+    '''
+    pattern = '|'.join(id_list)
+    df = data.loc[data[id_col].str.contains(pattern, case=False)]
+    return df
+
+
+def get_common_burst_ids(data):
+    '''
+    Get list of burst IDs common among all processed dates
+    Parameters:
+    ----------
+    data: pandas.DataFrame
+      Dataframe containing info for stitching (e.g. burst IDs)
```

Comment:
```suggestion

    Parameters:
    ----------
    data: pandas.DataFrame
      Dataframe containing info for stitching (e.g. burst IDs)
      
```
Extra lines for readability.